bookkeeper-server/src/test/java/org/apache/bookkeeper/proto/TestPerChannelBookieClient.java

```diff
@@ -314,7 +314,7 @@ public void testEpollChannelTcpUserTimeout() throws Exception {
         EventLoopGroup eventLoopGroup = new EpollEventLoopGroup();
         OrderedExecutor executor = getOrderedSafeExecutor();
         ClientConfiguration conf = new ClientConfiguration();
-        int tcpUserTimeout = 1234;
+        int tcpUserTimeout = 1236; // this value may be rounded on some Linux implementations
```

Comment:
Out of curiosity, is it documented anywhere? How did you find that this rounding is going, @eolivelli ?

Reply:
Some references

https://datatracker.ietf.org/doc/html/rfc5482

I didn't find more

Reply:
Sure, that's the RFC, but where did you see that some linux implementations round it?

Reply:
I have this setup:

> mvn -v
> Maven home: /usr/share/maven
> Java version: 1.8.0_292, vendor: Private Build, runtime: /usr/lib/jvm/java-8-openjdk-amd64/jre
> Default locale: en, platform encoding: UTF-8
> OS name: "linux", version: "4.15.0-64-generic", arch: "amd64", family: "unix"

>  uname -a
> Linux jenkins-pulsar-basic-runner-324 4.15.0-64-generic #73-Ubuntu SMP Thu Sep 12 13:16:13 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

The problem always reproduces on our Jenkins slaves at DataStax, basically every test run starts a new VM, installs Ubuntu and Java and then builds bookkeeper from the master ASF branch.
it is 100% reproducible on those machines.

I had not time to investigate, because I should learn more about the machines.
We should check that kernel version + Java version....

Reply:
@eolivelli do you know if with this change the test is passing consistently in your environment?

Reply:
@eolivelli this smells like an endianness problem somewhere rather than a rounding problem. See https://github.com/jnr/jnr-unixsocket/pull/55 for example. Did it start at some BK version (dependencies updates etc) or it was the first run on these VMs?

Reply:
Another suspect is that [TCP_USER_TIMEOUT](https://man7.org/linux/man-pages/man7/tcp.7.html) (tcpUserTimeout feeds into it) is `unsigned int` but netty's interface to jni passes it as it is jint: https://github.com/netty/netty/blob/23405e2000427e9cad104913e7071d8d08e91a3c/transport-native-epoll/src/main/c/netty_epoll_linuxsocket.c#L144-L146

i.e. for multicast netty casts parameter to `u_int`: https://github.com/netty/netty/blob/23405e2000427e9cad104913e7071d8d08e91a3c/transport-native-epoll/src/main/c/netty_epoll_linuxsocket.c#L78-L81

Reply:
The test was born a few weeks ago.
It passes on GH CI and it fails on Datastax CI.

Let me paste the results of the tests on this branch

Reply:
@dlg99 is it possible that 1234 is corrupted to 1236 due an endianness problem ?
but 1236 is not changed to another value

Reply:
TBH I do not know.
I experimented with endianness conversion, neither 1234 nor 1236 seem problematic or radically different, even if I add casting to unsigned int.
OTOH, TCP_USER_TIMEOUT is defined as a 15bit unsigned in the RFC but I don't think 15 vs 16bit matters for either value. 

but extra 2ms timeout making the test pass makes even less sense

Reply:
The other cure for this problem is to revert the change at all.

@RaulGracia said that in the test environments this feature is a good enhancement and fixes a problem.

So probably we can change the test to set a value that is likely to be used in production.
If Netty/Linux or other parts om the stack do not preserve every value it is not a BK problem.
We can open a ticket on Netty, I am pretty sure that we cannot so anything in BK.

I want to fix this test that is consistently failing and it is blocking some CI pipeline.